client/src/app/projects/components/project-item/project-item.component.spec.ts

```diff
@@ -1,35 +1,77 @@
 import { CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';
 import { ComponentFixture, TestBed } from '@angular/core/testing';
 import { MatDialog } from '@angular/material/dialog';
+import { Router } from '@angular/router';
 import { RouterTestingModule } from '@angular/router/testing';
+import { EMPTY } from 'rxjs';
 import { TruncatePipe } from 'src/app/shared/pipes/truncate.pipe';
-import { ProjectInterfaceMock } from '../../services/mocks/project-interface.mock';
+import { ProjectMock } from '../../services/mocks/project.mock';
 
 import { ProjectItemComponent } from './project-item.component';
 
 describe('ProjectItemComponent', () => {
   let component: ProjectItemComponent;
   let fixture: ComponentFixture<ProjectItemComponent>;
+  let dialog: MatDialog;
+  let router: Router;
 
   beforeEach(async () => {
     await TestBed.configureTestingModule({
       imports: [RouterTestingModule],
       declarations: [ProjectItemComponent, TruncatePipe],
       providers: [
-        { provide: MatDialog, useValue: { open: jasmine.createSpy() } },
+        {
+          provide: MatDialog,
+          useValue: {
```

Comment:
I guess that we can move this code in separate file. We have it in several places